packages/js/product-editor/src/blocks/product-fields/product-details-section-description/edit.tsx

```diff
@@ -0,0 +1,229 @@
+				);
+				return;
+			}
+
+			const newProduct = await ( saveEntityRecord(
+				'postType',
+				'product',
+				{
+					title: AUTO_DRAFT_NAME,
+					status: 'auto-draft',
+					...productTemplate.product_data,
+				}
+			) as never as Promise< Product > );
+
+			const url = getNewPath( {}, `/product/${ newProduct.id }` );
+			navigateTo( { url } );
+
+			onClose();
+		};
+	}
+
+	return (
+		<Fill name={ rootClientId }>
+			<div { ...blockProps }>
+				<p>
+					{ createInterpolateElement(
+						/* translators: <ProductType />: the product type. */
+						__(
+							'This is a <ProductType /> product.',
+							'woocommerce'
+						),
+						{
+							ProductType: <Fragment>{ productType }</Fragment>,
+						}
+					) }
+				</p>
+
+				<Dropdown
+					renderToggle={ ( { isOpen, onToggle } ) => (
+						<Button
+							aria-expanded={ isOpen }
+							variant="link"
+							onClick={ onToggle }
+						>
+							<span>
+								{ __( 'Change product type', 'woocommerce' ) }
+							</span>
+						</Button>
+					) }
+					renderContent={ ( { onClose } ) => (
+						<div className="components-dropdown-menu__menu">
+							<MenuGroup>
+								{ supportedProductTemplates.map(
+									( productTemplate ) => (
+										<MenuItem
+											key={ productTemplate.id }
+											info={
+												productTemplate.description ??
+												undefined
+											}
+											onClick={ menuItemClickHandler(
+												productTemplate,
+												onClose
+											) }
+										>
+											{ productTemplate.title }
+										</MenuItem>
+									)
+								) }
+							</MenuGroup>
+
+							{ unsupportedProductTemplates.length > 0 && (
+								<MenuGroup>
+									<Dropdown
+										// @ts-ignore Property does exists
+										popoverProps={ {
+											placement: 'right-start',
+										} }
+										renderToggle={ ( {
+											isOpen,
+											onToggle,
+										} ) => (
+											<MenuItem
+												aria-expanded={ isOpen }
+												icon={ chevronRight }
+												iconPosition="right"
+												onClick={ onToggle }
+											>
+												<span>
+													{ __(
+														'More',
+														'woocommerce'
+													) }
+												</span>
+											</MenuItem>
+										) }
+										renderContent={ () => (
+											<div className="components-dropdown-menu__menu">
+												<MenuGroup>
+													{ unsupportedProductTemplates.map(
+														( productTemplate ) => (
+															<MenuItem
+																key={
+																	productTemplate.id
+																}
+																info={
+																	productTemplate.description ??
+																	undefined
+																}
+																onClick={ menuItemClickHandler(
+																	productTemplate,
+																	onClose
+																) }
+															>
+																{
+																	productTemplate.title
+																}
+															</MenuItem>
+														)
```

Comment:
It would be nice to avoid the duplication of this.

Reply:
done